lib/worker-pipeline.js

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2021 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+'use strict';
+
+const debug = console.log;
+const { Engine, Plan } = require('@nui/asset-compute-pipeline');
+const WorkerCallbackTransformer = require('./worker-transformer');
+
+const fs = require('fs-extra');
+const path = require('path');
+
+/**
+ * Represents a pipeline build around a worker callback.
+ * Mostly here for backwards compatibility.
+ */
+class AssetComputeWorkerPipeline {
+    constructor(renditionCallback, options={}){
+        this.options = options;
+
+        this.transformers = [];
+        if(this.options.transformerCatalog){
+            debug("Adding transformers from transformer catalog to the pipeline");
+            this.transformers = this.transformers.concat(options.transformerCatalog);
+        }
+
+        this.buildTransformersFromManifests(renditionCallback);
+        debug("Built worker transformer using rendition callback 'renditionCallback'");
+    }
+
+    /**
+     * Builds transformers from manifests defined in the options object
+     * @param {*} renditionCallback worker rendition callback to use in the built transformer
+     * @param {*} options options, must contain manifests (if not - will not create any transformer)
+     * @returns a list of transformers available for this pipeline instance
+     */
+    buildTransformersFromManifests(renditionCallback) {
+        if (!this.options.manifests) {
+            debug("No additional manifest in the pipeline options, no additional WorkerCallbackTransformer added to pipeline");
+            return this.transformers;
+        }
+
+        const manifests = this.options.manifests;
+        debug('Manifest array', manifests);
+    
+        const transformers = [];
+        manifests.forEach(manifest => {
+            const transformer = new WorkerCallbackTransformer(renditionCallback, manifest);
+
+            debug(`Created a transformer named ${transformer.name} with manifest:`);
+            debug(manifest);
+
+            transformers.push(transformer);
+        });
+
+        debug(`Adding ${transformers.length} WorkerCallbackTransformer to the pipeline`);
+        this.transformers = transformers.concat(this.options.transformerCatalog);
```

Comment:
no need to concat with transformerCatalog here since you did that in the contructor

Reply:
Thanks, good catch. Missed some code removal when refactoring.